hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/util/resource/DominantResourceCalculator.java

```diff
@@ -118,8 +121,10 @@ public int computeAvailableContainers(Resource available, Resource required) {
      * If both are 0, we return 0 which will skip scheduling on both code paths.
      */
     if (required.getMemory() == 0) {
-      if (required.getVirtualCores() == 0)
+      if (required.getVirtualCores() == 0) {
+        LOG.warn("Got a resources reservation with 0 vcore and 0 memory, acting as if no resource was available");
```

Comment:
We will only get a warn and not much context. Maybe we could verify memory and vcores before entering computeAvailableContainers / or use exception based mechanism to log more info about the request that led to this -> application_id and maybe the complete request.

Reply:
Agree, that's a good idea since there is actually only one relevant path